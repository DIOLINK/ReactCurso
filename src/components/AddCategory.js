import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((category) => [inputValue, ...category]);
      setInputValue('');
    }
  };

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
