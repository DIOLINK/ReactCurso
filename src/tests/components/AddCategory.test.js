import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Probando AddCategory.js', () => {
  const setCategories = jest.fn();

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });
  test('Debe renderizar correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Debe de cambiar la caja de texto', () => {
    const value = 'Hola Roger';
    const input = wrapper.find('input');
    input.simulate('change', { target: { value } });
    expect(wrapper.find('input').prop('value')).toBe(value);
  });
  test('No debe postear nada', () => {
    const form = wrapper.find('form');
    form.simulate('submit', { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });
  test('Debe de llamar el setCategories y limpiar el input', () => {
    wrapper.find('input').simulate('change', { target: { value: 'Holaaaa' } });

    wrapper.find('form').simulate('submit', { preventDefault() {} });

    expect(setCategories).toHaveBeenCalled();
    // expect(setCategories).toHaveBeenCalledTimes(2);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(wrapper.find('input').prop('value')).toEqual('');
  });
});
