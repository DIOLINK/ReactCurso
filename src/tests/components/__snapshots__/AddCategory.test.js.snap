// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando AddCategory.js Debe renderizar correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
